client.go

```diff
@@ -62,7 +52,20 @@ type ClientInt interface {
 // information can be found in their respective documentation.
 // Numerous connection options may be specified by configuring a
 // and then supplying a ClientOptions type.
-type Client struct {
+
+type Client interface {
+	IsConnected() bool
+	Connect() Token
+	Disconnect(uint)
+	disconnect()
```

Comment:
The method was already there in the previous interface. I think there shouldn't be any issue to move that method from the interface to the struct itself.